src/common/reactNativeProjectHelper.ts

```diff
@@ -10,53 +10,112 @@ import {Package} from "./node/package";
 import {ErrorHelper} from "../common/error/errorHelper";
 import {InternalErrorCode} from "../common/error/internalErrorCode";
 
+export interface ParsedPackageName {
+    packageName: string;
+    useSemverCoerce: boolean;
+}
+
+export interface RNPackageVersions {
+    reactNativeVersion: string;
+    reactNativeWindowsVersion: string;
+}
+
 export class ReactNativeProjectHelper {
 
     public static getRNVersionsWithBrokenMetroBundler() {
         // https://github.com/Microsoft/vscode-react-native/issues/660 for details
         return ["0.54.0", "0.54.1", "0.54.2", "0.54.3", "0.54.4"];
     }
 
-    public static getReactNativeVersion(projectRoot: string): Q.Promise<string> {
-        return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(projectRoot)
+    public static getReactNativeVersions(projectRoot: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        return ReactNativeProjectHelper.getReactNativePackageVersionsFromNodeModules(projectRoot, isRNWindows)
             .catch(err => {
-                return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(projectRoot);
+                return ReactNativeProjectHelper.getReactNativeVersionsFromProjectPackage(projectRoot, isRNWindows);
             });
     }
 
-    public static getReactNativePackageVersionFromNodeModules(projectRoot: string): Q.Promise<string> {
-        return new Package(projectRoot).dependencyPackage("react-native").version()
-            .then(version => ReactNativeProjectHelper.processVersion(version))
-            .catch(err => {
-                throw ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
-            });
+    public static getReactNativePackageVersionsFromNodeModules(projectRoot: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        let versionPromises: Q.Promise<string>[] = [];
+
+        versionPromises.push(
+            new Package(projectRoot).getPackageVersionFromNodeModules("react-native")
+                .then(version => ReactNativeProjectHelper.processVersion(version, true))
+                .catch(err => {
+                    throw ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
+                })
+            );
+
+        if (isRNWindows) {
+            versionPromises.push(
+                new Package(projectRoot).getPackageVersionFromNodeModules("react-native-windows")
+                    .then(version => ReactNativeProjectHelper.processVersion(version, false))
+                    .catch(err => "")
+            );
+        }
+
+        return Q.all(versionPromises).then(packageVersions => ({
+            reactNativeVersion: packageVersions[0] || "",
```

Comment:
Please try to avoid constant indices because they can lead to the issues that are hard debuggable in such context.